scripts/west_commands/runners/intel_adsp.py

```
@@ -0,0 +1,118 @@
+        parser.add_argument('-t', '--timeout', type=int, default=60,
+                            help='the pty where stdout goes to')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return IntelAdspBinaryRunner(cfg,
+                                    remote_host=args.remote_host,
+                                    timeout=args.timeout,
+                                    )
+
+    def do_run(self, command, **kwargs):
+        self.logger.info('Starting Intel ADSP runner')
+
+        if command == 'flash':
+            if re.search("intel_adsp_cavs", self.platform):
+                self.require(self.cavstool)
+                self.flash(**kwargs)
+            else:
+                self.run_sim(**kwargs)
+        else:
+            self.logger.info('Only support flash cmd now. Do nothing...')
+
+    def flash(self, **kwargs):
+        # Copy the zephyr and cavstool.py to target remote ADSP host
+        download_cmd = (['scp', f'{self.cavstool}', f'{self.bin_fw}', f'{self.remote_host}:'])
+        self.logger.info(f"cavstool({self.cavstool}), fw('{self.bin_fw})")
+
+        try:
+            self.check_call(download_cmd)
+        except subprocess.CalledProcessError:
+            return 1
+
+        # Excute the command "sudo ./cavstool.py zephyr.ri"
+        # FIXME: Make the cavstool.py works as a service and don't need
+        # to restart every time.
+        self.run_cmd = (['ssh', f'{self.remote_host}', 'sudo', './cavstool.py', 'zephyr.ri', '-t', f'{self.timeout}'])
+        self.kill_cmd = (['ssh', f'{self.remote_host}', 'sudo', 'pkill' , '-9', '-f', 'cavstool'])
+
+        self.logger.info(f"rcmd: {self.run_cmd}")
+
+        # We need this to cleanup the remote host process.
+        signal.signal(signal.SIGTERM, self.do_cleanup)
+
+        try:
+            self.check_call(self.run_cmd)
+        except Exception as e:
+            self.logger.error(e)
+            self.do_cleanup()
+            return 1
```

Please don't swallow exceptions like this.

---

Thanks for correcting this. Fix it.